packages/esm-patient-chart-app/src/clinical-views/hooks/useLastEncounter.ts

```diff
@@ -0,0 +1,27 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { type Encounter } from '../types';
+import useSWRImmutable from 'swr/immutable';
+
+const encounterRepresentation =
+  'custom:(uuid,encounterDatetime,encounterType,location:(uuid,name),' +
+  'patient:(uuid,display,age,identifiers,person),encounterProviders:(uuid,provider:(uuid,name)),' +
+  'obs:(uuid,obsDatetime,voided,groupMembers,concept:(uuid,display,name:(uuid,name)),value:(uuid,name:(uuid,name,display),' +
+  'names:(uuid,conceptNameType,name,display))),form:(uuid,name))';
+
+export function useLastEncounter(patientUuid: string, encounterType: string) {
+  const query = `encounterType=${encounterType}&patient=${patientUuid}&limit=1&order=desc&startIndex=0`;
+  const endpointUrl =
+    patientUuid && encounterType ? `/ws/rest/v1/encounter?${query}&v=${encounterRepresentation}` : null;
+
+  const { data, error, isLoading, isValidating } = useSWRImmutable<{ data: { results: Array<Encounter> } }, Error>(
```

Comment:
This seems like an incorrect use of `useSWRImmutable()`. Please just `useSWR()` here. We can adjust the SWR config if it needs to not reload under some circumstances or something.